app/src/main/java/org/imozerov/streetartview/storage/DataSource.kt

```diff
@@ -0,0 +1,60 @@
+package org.imozerov.streetartview.storage
+
+import android.os.SystemClock
+import io.realm.Realm
+import io.realm.RealmList
+import io.realm.RealmObject
+import org.imozerov.streetartview.storage.model.RealmArtObject
+import org.imozerov.streetartview.storage.model.RealmAuthor
+import org.imozerov.streetartview.storage.model.RealmString
+import org.imozerov.streetartview.ui.model.ArtObjectUi
+import rx.Observable
+import java.util.*
+
+/**
+ * Created by imozerov on 05.02.16.
+ */
+class DataSource(internal var realm: Realm) {
+
+    fun listArtObjects(): Observable<List<ArtObjectUi>> {
+        return realm
+                .allObjects(RealmArtObject::class.java)
+                .asObservable()
+                .cache()
+                .map {
+                    val listOfArtObjects = ArrayList<ArtObjectUi>(it.size)
+                    for (model in it) {
+                        listOfArtObjects.add(ArtObjectUi(model))
+                    }
+                    listOfArtObjects
+                }
+    }
+
+    fun getArtObject(id: String): Observable<ArtObjectUi> {
+        return realm
+                .where(RealmArtObject::class.java)
+                .equalTo("id", id)
+                .findFirst()
+                .asObservable<RealmObject>()
```

Comment:
Compiler failed to detect type of findFirst return object. We need to help him do it.